### PR TITLE
Sort list of downloads, and add link to mpdecimal sources

### PR DESCRIPTION
**Pull request by erlend-aasland**
- python/cpython#115119

### CODE REVIEW

**Comment by erlend-aasland** [comment]
I also have a `mpdecimal` branch[^1] with the 2.5.1 and 4.0.0 sources; let me know if you want me to push that:

```
$ git log mpdecimal
commit 48316ec025c1ebe500854c332be0a12c640c7301 (HEAD -> mpdecimal, tag: mpdecimal-4.0.0)
Author: Erlend E. Aasland <erlend@python.org>
Date:   Thu Feb 8 10:07:09 2024 +0100

    Add mpdecimal 4.0.0 sources
    
    Signed-off-by: Erlend E. Aasland <erlend@python.org>

commit 5da6848fbab6bb6dc3aa09646c313fa4d3b5f522 (tag: mpdecimal-2.5.1)
Author: Erlend E. Aasland <erlend@python.org>
Date:   Thu Feb 8 09:59:09 2024 +0100

    Add mpdecimal 2.5.1 sources
    
    Signed-off-by: Erlend E. Aasland <erlend@python.org>

commit 5110f51f5d5966683eaa43375bb0f3716cf312e6
Author: Erlend E. Aasland <erlend@python.org>
Date:   Thu Feb 8 09:58:02 2024 +0100

    README: add recipe for how to update mpdecimal sources

commit d239d6305768c92ca842e5b814bab80f24dc2ad0
Author: Zachary Ware <zachary.ware@gmail.com>
Date:   Mon May 22 10:43:20 2017 -0700

    Create README.md

```


[^1]: branched off of the initial commit

File: README.rst [hunk 1]
```diff
@@ -50,13 +50,14 @@ the release management team to create those builds.
...
 --------------
 
-- ``sqlite``: https://www.sqlite.org/download.html
 - ``bzip2``: https://www.sourceware.org/bzip2/downloads.html
+- ``libffi`` : https://github.com/libffi/libffi
+- ``mpdecimal`` : https://www.bytereef.org/mpdecimal/download.html
+- ``openssl``: https://www.openssl.org/source/
+- ``sqlite``: https://www.sqlite.org/download.html
+- ``tcl``/``tk``: https://tcl.tk/software/tcltk/download.html
 - ``xz``: https://tukaani.org/xz/
```
Comment (hugovk) [inline README.rst]: Can update this to its redirect:
```suggestion
- ``xz``: https://xz.tukaani.org/xz-utils/
```

**Review by zware — APPROVED** [review]
LGTM!

**Comment by zware** [comment]
Please go ahead and push the `mpdecimal` branch and tags as well.

**Comment by erlend-aasland** [comment]
> Please go ahead and push the `mpdecimal` branch and tags as well.

Done: https://github.com/python/cpython-source-deps/tree/mpdecimal

BTW, we just hit 100 tags! 💯

**Comment by zware** [comment]
Thanks @erlend-aasland!

Just to record the action *somewhere*, I also adjusted branch protection rules to prevent deletion of `mpdecimal` and `libffi`, and removed some dead branches (`openssl`, `openssl-1.0.2`, `openssl-1.1.1`, `tk85`, `tcl85`, `bsddb`, and `tix`).  The tags from the old branches are still there, and there were no un-tagged commits on those branches.

**Pull request merged**